support/builder/build_dep_populate.sh

```diff
@@ -0,0 +1,463 @@
+            setBucket
+        fi
+    fi
+}
+
+checkBucket() {
+    region="${1}"
+    bucket="${2}"
+    aws --endpoint-url "${region}" s3api list-objects --bucket "${bucket}"
+}
+
+setRegion() {
+    # Sets the region used by the bucket
+    if [ "${s3type}" == 'minio' ]; then
+        sgroups=("default" "dev" "prod" "acceptance" "live" "blue" "green")
+        for i in "${sgroups[@]}"; do
+            if curl -s localhost:9631/services/builder-minio/"${i}" > /dev/null; then
+                minioIP=$(curl -s localhost:9631/services/builder-minio/"${i}" | jq .sys.ip)
+                echo ""
+                echo "We've detected your minio instance at: ${minioIP}!"
+                echo ""
+                read -r -p "Would you like to use this minio instance? [y/N] " response
+                if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+                    echo "Setting endpoint to ${minioIP//\"}:9000"
+                    if curl -s localhost:9631/services/builder-minio/"${i}" | jq .cfg.use_ssl | grep "true"> /dev/null; then
+                        export AWS_REGION="https://${minioIP//\"}:9000"
+                    else
+                        export AWS_REGION="http://${minioIP//\"}:9000"
+                    fi
+                    return
+                else
+                    echo ""
+                    echo "==========================================================="
+                    echo "Please enter the minio endpoint URI and press [ENTER]:"
+                    echo "(http://localhost:9000 || https://10.1.250.4:9000)"
+                    echo "==========================================================="
+                    read -r region_name
+                    export AWS_REGION=${region_name}
+                    return
+                fi
+            fi
+        done
+    else
+        echo ""
+        echo "==========================================================="
+        echo "Please enter the region for your bucket and press [ENTER]:"
+        echo "(us-west-1 us-east-1 us-west-2 etc )"
+        echo "==========================================================="
+        read -r region_name
+        export AWS_REGION=${region_name}
+    fi
+}
+
+failed_uploads=()
+uploadHarts() {
+    # Takes the artifact array generated via
+    # traversing the data path and uploads to s3
+    uploadMsg
+    putArtifacts
+    echo ""
+    echo "########################################"
+    echo "${#artifacts[@]} Artifacts have been ingested!"
+    echo "########################################"
+    echo ""
+
+    if (( "${failed_uploads[@]}" )); then
+    echo ""
+    echo "########################################"
+    echo "The following artifacts failed on upload:"
+    for failure in "${failed_uploads[@]}"; do
+        echo "${failure}"
+    done
+    echo "########################################"
+    fi
+}
+
+putArtifacts() {
+    # shellcheck disable=SC2044
```

Comment:
I know it's kind of ugly, but considering that `/hab/tmp/harts/` was populated via a `s3 cp --recursive`, the possibility of unexpected files that could screw this up is real. Why not go with the safer version:
```bash
    while IFS= read -r -d '' file; do
        …
    done < <(find /hab/tmp/harts/ -name '*.hart' -print0)
```
?

Reply:
I dont see a case where anything problematic comes out of this loop as is. Any unexpected files that aren't harts would be ignored, any unexpected files that ARE harts would just be re-uploaded (which isn't really a problem) and any failures due to unexpected harts like for example some windows packages, would get logged.

Reply:
The problem would occur if there were a match on a file that contained whitespace or glob characters (anywhere in the path). I agree that *probably* won't happen, but since it's straightforward (albeit a tad ugly) to do the right thing, I though it was worth suggesting.